### DISCUSSION TITLE
Looking for detailed instructions to uninstall Music Assistant completely from Home Assistant

### DISCUSSION THREAD

**Discussion by spudje** [music-assistant/support#2033] [Q&A] [answered]
Just tried Music Assistant, but is does not offer what I'm looking for. So I want to completely remove it from my Home Assistant setup. Unfortunately I can't find any instructions for this in the documentation. So can someone write (and also add it to the documentation on https://music-assistant.io/) a step by step guide to completely emove everything from Home Assistant:
- Plugins, providers
- Media player devices/entities
- Add-on
- HACS component
- ...

Thanks!

**Comment by OzGav** [comment]
I don’t believe we need to as it is just like any other custom integration. Remove it from HA settings, remove it from HACS and restart HA

**Comment by spudje** [comment]
OK, I'll try that. Will that also remove the Add-on and all the media players?

**Reply by OzGav** [reply]
It will not remove the addon. You just remove that like other addons by clicking on the uninstall button. The media players will be removed when the integration is removed. If anything is left that will be a HA or HACS problem.

**Reply by spudje** [reply]
Seems to have worked fine and it also removes the addon with it. So no need to remove that separately. The only entity remaining is this one, as far as I can see: update.music_assistant_beta_update

**Reply by OzGav** [reply]
That is created by HA. Try a HA restart and see if that is removed automatically. If not try and delete it from HA SETTINGS>>DEVICES AND SERVICES>>ENTITIES If you can't then you will have to ask in the HA forums.

**Reply by Roving-Ronin** [reply]
Interested in this also, as just installed it on 2 x HA instances and wasn't suited to my needs so uninstalled. 
Removed in HACS and restarted and rebooted HA, but then had notifications that there was a new Update of MA to install. Checked and there's the entities still remaining:

![image](https://github.com/music-assistant/hass-music-assistant/assets/108674933/6209f6f7-0443-4d26-9b27-5d49fe85518f)

Went to remove them from ./storage/core.config_entries and find they aren't listed there. Tried calling them in the Purge Entities service and if you can even define the (they don't appear in the list, so manually typed their full ID in) also does nothing. Purged browser cache and still there. Lastly going to Developer Tools --> Statistics to 'Fix' and delete them there, finds them not visible/listed to remove.

Would suggest that DEVs should put a warning about this long standing issue at their top of the Repo/HACS page, to warn people prior to installing it. First Add-On in 3 years of HA to not remove cleanly.

**Reply by OzGav** [reply]
It's not a long standing issue for MA. Look at your image. Those entities were created by the supervisor. Have you removed the addon as well?

**Reply by Roving-Ronin** [reply]
The addon was removed via HACS, also confirmed removed from customs_components etc.

If it's the supervisor how to remove, as I've searched and followed all the suggestions with no success.

**Reply by OzGav** [reply]
The addon is not installed by HACS. You are not understanding the difference between an addon and a custom component. Go to HA SETTINGS >> ADDONS and I expect you will still see the server there. Click on UNINSTALL

**Reply by marcelveldt** [reply]
That is not entirely true. If you installed the HA integration on HAOS, it will also install (and manage!) the add-on. That is where that checkbox "Use the official add-on" is for. 

In that case the HA integration will manage the add-on completely: 

- if you install the HA integration, it will also install the add-on.
- it will auto update the add-on if that is needed.
- it will make sure that the add-on is running as long as the HA integration is enabled.
- if you disable the HA integration, it will shutdown the add-on. 
- if you uninstall the add-on, it will uninstall the add-on as well

So yeah, users that installed MA by starting with the HA integration from HACS with that checkbox selected, will not have to deal with the add-on in normal circumstances.

**Comment by marcelveldt** [comment]
> Would suggest that DEVs should put a warning about this long standing issue at their top of the Repo/HACS page, to warn people prior to installing it. First Add-On in 3 years of HA to not remove cleanly.

Long standing issue ? I have never ever heard of this before. This more likely looks like a supervisor issue tbh and not MA if you are sure the add-on is removed.

**Reply by marcelveldt** [reply, marked as answer]
For the record; the entities in the screenshot of @Roving-Ronin are coming from the HA supervisor. We do not have any direct control over them. So if these entities do not clean themselves, it's a supervisor issue, NOT a MA issue.

**Comment by marcelveldt** [comment]
@spudje @Roving-Ronin purely out of interest (so we can maybe improve docs or improve features) - what what the reason you installed MA in the first place (what did you expect to do for you) and what was it that you decided it was not what you were looking for ?

**Reply by MarkGTI** [reply]
How on earth do I uninstall this thing?

**Reply by OzGav** [reply]
You are going to have to be more specific
